app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CategoriesController < ApplicationController
+
+	def index
+		@categories = Category.all
+	end
+	
+	def new
+		@category = Category.new
+	end
+
+	def create
+		@category = Category.new permitted_params
+		if @category.save
+			redirect_to categories_path
+		else
+			render :new
+		end
+	end
+
+	def edit
+		@category = Category.find params[:id]
+	end
+
+	def update
+		@category = Category.find params[:id]
+		if @category.update permitted_params
+			redirect_to categories_path
+		else
+			render :edit
+		end
+	end
+
+	def show
+		
+	end
+
+	def destroy
+	 	@category = Category.find params[:id]
+	 	if @category.destroy
+	 		redirect_to categories_path
+	 	else
+	 		redirect_to categories_path
+	 	end
+	end 
+
+	private
+		def permitted_params
+			params.require(:category).permit(:title, :image)
+		end
+end
```

Comment:
Add flash messages for creating, updating and destroying a record. 

[http://api.rubyonrails.org/classes/ActionDispatch/Flash.html]